Server/JokesOnYou.Web.Api/Repositories/TagRepository.cs

```diff
@@ -36,13 +62,12 @@ public TagRepository(DataContext context, IMapper mapper)
                    join userJokeTags in _context.UserJokeTags
                    on tags.Id equals userJokeTags.TagId
                    where userJokeTags.JokeId == jokeId
-                   select new Tag 
+                   select new Tag
                    {
-                       
-                       Id = tags.Id, 
-                       Name = tags.Name,  
-                       Created = tags.Created, 
-                       OwnerId = tags.OwnerId, 
+                       Id = tags.Id,
+                       Name = tags.Name,
+                       Created = tags.Created,
+                       OwnerId = tags.OwnerId,
                        Likes = userJokeTags.Likes
                    }).ProjectTo<TagReplyDto>(_mapper.ConfigurationProvider).ToListAsync();
```

Comment:
We only need projectto or select. They do the same thing

Reply:
I Don't know about it. My friend told me he forgot to clear a empty line.so he told me to clear this space

Reply:
Select is needed here as we need to take Likes from userJokeTags and add into Tag. Then we need to map Tag to TagReplyDto

Reply:
You can do both in the .select, like i said they do the same thing. They map properties from one object to another. EF will hopefully in this case ignore that there are two .Select statements and instead combine them but probably want to double check the generated sql.

Reply:
Got your point. I think it will be,
`        public async Task<IEnumerable<TagReplyDto>> GetTagDtosByJokeIdAsync(int jokeId) =>
            await (from tags in _context.Tags
                   join userJokeTags in _context.UserJokeTags
                   on tags.Id equals userJokeTags.TagId
                   where userJokeTags.JokeId == jokeId
                   select new TagReplyDto 
                   {                      
                       Id = tags.Id, 
                       Name = tags.Name,  
                       Created = tags.Created, 
                       OwnerId = tags.OwnerId, 
                       Likes = userJokeTags.Likes
                   }).ToListAsync();`

Reply:
Yup something like that, want to stick with method syntax tho

Reply:
Looks more complex in method syntex I think. Previous Version looked simpler.

` public async Task<IEnumerable<TagReplyDto>> GetTagDtosByJokeIdAsync2(int jokeId) =>
            await (_context.Tags.
            Join(_context.UserJokeTags, 
                tag => tag.Id,
                userJokeTag => userJokeTag.TagId,
                (tag, userJokeTag) =>
                new
                {
                    tags = tag,
                    userJokeTags = userJokeTag
                }).
            Where(dto => dto.userJokeTags.JokeId == jokeId).
            Select(x =>
            new TagReplyDto
            {
                Id = x.tags.Id,
                Name = x.tags.Name,
                Created = x.tags.Created,
                OwnerId = x.tags.OwnerId,
                Likes = x.userJokeTags.Likes
            })).ToListAsync();`

Reply:
Btw, feedback if it is okay?

Reply:
I agree if we want to do a join here instead of subquery then query syntax looks more readable. @HellHunterMax might have some input on this

Reply:
I agree query syntax in this instance is more readable.